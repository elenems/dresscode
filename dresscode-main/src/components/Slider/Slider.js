import React, { PureComponent } from "react";
import AliceCarousel from "react-alice-carousel";
import "react-alice-carousel/lib/alice-carousel.css";
import Icon from "../../utils/Icon";
import PropTypes from "prop-types";

const res = {
  0: {
    items: 1
  },
  500: {
    items: 2
  },
  768:{
    items: 3
  },
  1024: {
    items: 4
  },
  1920: {
    items: 5
  }
};

class Slider extends PureComponent {
  state = {
    currentIndex: 0
  };

  onSlideChanged = e => this.setState({ currentIndex: e.item });

  slideNext = () =>
    this.setState({ currentIndex: this.state.currentIndex + 1 });

  slidePrev = () =>
    this.setState({ currentIndex: this.state.currentIndex - 1 });

  render() {
    return this.props.items.length ? (
      <div style={{ position: "relative" }}>
        <AliceCarousel
          items={this.props.items}
          buttonsDisabled={true}
          dotsDisabled={true}
          onSlideChanged={this.onSlideChanged}
          slideToIndex={this.state.currentIndex}
          responsive={res}
          mouseDragEnabled
        />
        <div className="alice-carousel__prev-btn">
          <button
            className="alice-carousel__prev-btn-wrapper"
            onClick={() => this.slidePrev()}
          >
            <Icon color="white" name="left-arrow" />
          </button>
        </div>

        <div className="alice-carousel__next-btn">
          <button
            className="alice-carousel__next-btn-wrapper"
            onClick={() => this.slideNext()}
          >
            <Icon color="white" name="right-arrow" />
          </button>
        </div>
      </div>
    ) : null;
  }
}

Slider.propTypes = {
  items: PropTypes.array
};

export default Slider;
