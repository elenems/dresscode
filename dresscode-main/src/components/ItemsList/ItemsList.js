import React, { Component } from "react";
import axios from "axios";
import ListItem from "./ListItem/ListItem";
import PropTypes from "prop-types";
import Slider from "../Slider/Slider";
export default class ItemsList extends Component {
  state = {
    items: []
  };

  componentDidMount() {
    const { limit, fetchLink } = this.props;
    axios
      .get(`${fetchLink}?limit=${limit}`)
      .then(data => {
        this.setState({ items: data.data.items });
      })
  }

  render() {
    const items = this.state.items.map(item => {
      return <ListItem user={this.props.user} key={item.id} item={item} />;
    });

    return (
      <div className="items-list">
        <Slider items={items} />
      </div>
    );
  }
}

ItemsList.propTypes = {
  limit: PropTypes.number.isRequired,
  fetchLink: PropTypes.string.isRequired
};
