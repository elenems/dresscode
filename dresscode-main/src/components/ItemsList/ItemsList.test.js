import { shallow } from "enzyme";
import React from "react";
import ItemsList from "./ItemsList";
import Slider from "../Slider/Slider";

describe("ItemsList", () => {
  const apiLink = "https://europe-west2-dresscode-691e5.cloudfunctions.net/api";
  const props = {
    limit: 5,
    fetchLink: `${apiLink}/getPopularClothes`,
    user: {}
  };

  it("Matches snapshot", () => {
    const wrapper = shallow(<ItemsList {...props} />).debug();
    expect(wrapper).toMatchSnapshot();
  });

  it("Renders atleast one Item if state has items", () => {
    const wrapper = shallow(<ItemsList {...props} />);
    wrapper.setState({
      items: [{ id: 1 }]
    });
    expect(wrapper.find(Slider).prop("items").length).toBeGreaterThan(0);
  });

  it("Sets received data from server", async () => {
    const wrapper = shallow(<ItemsList {...props} />);
    await wrapper.instance().componentDidMount();
    expect(wrapper.state("items").length).toBeGreaterThan(0);
  });
});
