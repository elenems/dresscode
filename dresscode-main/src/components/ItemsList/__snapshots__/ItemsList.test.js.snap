// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemsList Matches snapshot 1`] = `
"<div className=\\"items-list\\">
  <Slider items={{...}} />
</div>"
`;
